python/dgl/distributed/dist_graph.py

```diff
@@ -557,3 +599,80 @@ def _get_all_edata_names(self):
                 # Remove the prefix "edge:"
                 edata_names.append(name[5:])
         return edata_names
+
+def _get_part(elements, local_ids):
+    if isinstance(elements, DistTensor):
+        masks = elements[local_ids]
+        return F.boolean_mask(local_ids, masks)
+    else:
+        elements = utils.toindex(elements)
+        masks = F.gather_row(elements.tousertensor(), local_ids)
+        return F.boolean_mask(local_ids, masks)
+
+def node_split(nodes, partition_book, rank):
+    ''' Split nodes and return a subset for the local rank.
+
+    This function splits the input nodes based on the partition book and
+    returns a subset of nodes for the local rank. This method is used for
+    dividing workloads for distributed training.
+
+    The input nodes can be stored as a vector of masks. The length of the vector is
+    the same as the number of nodes in a graph; 1 indicates that the vertex in
+    the corresponding location exists.
+
+    Parameters
+    ----------
+    nodes : 1D tensor or DistTensor
+        A boolean mask vector that indicates input nodes.
+    partition_book : GraphPartitionBook
+        The graph partition book
+    rank : int
+        The rank of the current process
+
+    Returns
+    -------
+    1D-tensor
+        The vector of node Ids that belong to the rank.
+    '''
+    num_nodes = 0
+    for part in partition_book.metadata():
+        num_nodes += part['num_nodes']
```

Comment:
I remember there is a `total_num_nodes` field?

Reply:
it doesn't seem to have such a field. the metadata is stored in a list of dict. each dict has the information of the partition.